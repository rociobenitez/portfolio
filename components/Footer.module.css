.footer {
    width: 100%;  
    height: 10vh;
    text-align: center;
    display: flex;
    align-items: center;
    padding: 1em;
}

.container {
    display: flex;
    width: 100%;
    justify-content: space-between;
    align-items: center;
    color: var(--light-slate);
    font-family: var(--font-mono);
    font-size: var(--fz-xxs);
}

.divIcons{
    display: flex;
    justify-content: center;
    margin-top: 1em;
}

.divIcons a {
    margin: 0 .5em;
}
