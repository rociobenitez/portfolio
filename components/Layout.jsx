import PropTypes from 'prop-types';
import Head from 'next/head';
import { ThemeProvider } from "styled-components";
import { Nav, Footer } from '.';
import { GlobalStyle, theme } from '../styles';

export const Layout = ({ children, title, image, description, keywords }) => {

  const origin = (typeof window === 'undefined') ? '' : window.location.origin;

  return (
    <>
      <Head>
        <title>{ title }</title>
        <meta name = "description"              content = { description } />
        <meta name = "image"                    content = { image } />
        <meta name = "keywords"                 content = { keywords } />

        <meta property = "og:type"              content = "website" />
        <meta property = "og:title"             content = { title } />
        <meta property = "og:description"       content = { description } />
        <meta property = "og:site_name"         content = { title } />
        <meta property = "og:image"             content = { `${ origin }/assets/cover-portfolio-rociobenitez.png` } />

        <meta property = "twitter:card"         content = "summary" />
        <meta property = "twitter:title"        content = { title } />
        <meta property = "twitter:description"  content = { description } />

        <link rel="icon" href="/assets/ilustracion-rocio.png" />
      </Head>
      <ThemeProvider theme = { theme }>
        <GlobalStyle />
          <Nav />
          <main>
              { children }
          </main>
          <Footer />
      </ThemeProvider>
    </> 
  )
}

Layout.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string,
  description: PropTypes.string,
  keywords: PropTypes.string,
  image: PropTypes.string,
};

Layout.defaultProps = {
  title: 'Rocío Benítez',
  description: 'Este es el portfolio de Rocío Benítez',
  keywords: 'portfolio, react, nextjs, frontend, developer, frontend developer, javascript, projects',
  image: null,
};
