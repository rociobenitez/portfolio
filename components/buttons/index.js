export * from './Button'
export * from './SmallButton'