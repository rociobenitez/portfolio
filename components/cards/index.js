export * from './CardProject'
export * from './CardItems'
export * from './CardAbout'



