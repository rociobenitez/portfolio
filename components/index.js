export * from './buttons/index'
export * from './cards/index'
export * from './projects/index'
export * from './sections/index'

export * from './ActiveLink'
export * from './Footer'
export * from './GalleryAbout'
export * from './Layout'
export * from './Nav'



