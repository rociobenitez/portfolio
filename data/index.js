export * from './about'
export * from './links'
export * from './portfolio'
