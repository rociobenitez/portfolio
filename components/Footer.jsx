import React from 'react'
import Link from 'next/link';
import { AiFillGithub, AiFillLinkedin } from 'react-icons/ai'
import { linkedin, github } from '../data';
import styles from './Footer.module.css'

export const Footer = () => {
  return (
    <footer className={styles.footer}>
      <div className={styles.container}>
          <div>Designed &amp; Built by Roc√≠o</div>
          <div className={styles.divIcons}>
            <a href = { linkedin } target='_blank' rel='noreferrer'>
              <AiFillLinkedin size="2em"/>
            </a>
            <a href = { github } target='_blank' rel='noreferrer'>
              <AiFillGithub size="2em"/>
            </a>
          </div>
      </div>
    </footer>
  )
}
