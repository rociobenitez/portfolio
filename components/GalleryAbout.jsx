import { Card, Grid, Spacer } from '@nextui-org/react'
import { images } from '../data'

export const GalleryAbout = () => {
    return (
        <Grid.Container gap = {1} justify = "center">
            {
                images.map((item, index) => (
                    <Grid xs={3} sm={6} key={ index }>
                        <Card
                            isHoverable 
                            css={{
                                bg: "transparent",
                                w: "100%",
                                borderRadius: "var(--border-radius)",
                                '&:hover': {
                                    'img': {
                                        opacity: '.9',
                                        transform: 'scale(1.1)',
                                        transition: 'transform 500ms ease-out',
                                    },
                                    'div p': { opacity: '0' },
                                    'div h4' : { opacity: '0' }
                                }
                            }}
                        >
                        <Card.Image
                            src        = { '/assets/' + item.src }
                            alt        = { item.alt }
                            width      = '100%'
                            height     = { 100 }
                            objectFit  = 'cover'
                            css = {{
                                opacity: '.3' ,
                                transition: 'opacity 300ms ease'
                             }}  
                        />
                        </Card>
                    </Grid>
                ))
            }
            <Spacer y = {2}/>
        </Grid.Container>
    )
}
