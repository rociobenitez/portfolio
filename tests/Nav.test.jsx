import { render } from "@testing-library/react"
import { Nav } from "../components"
import { menuItems } from "../data"

describe('Tests in <Nav />', () => { 
    test('match with snapshot', () => { 
        const { container } = render( <Nav /> ); 
        expect( container ).toMatchSnapshot();
    })

    test('return array items', () => {
        const items = menuItems;
        expect( menuItems.length ).toBeGreaterThan(0);
        expect( menuItems[0] ).toEqual({
            text: expect.any( String ),
            href: expect.any( String ),
        })
    })
})