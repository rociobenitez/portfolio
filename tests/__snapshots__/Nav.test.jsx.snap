// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Nav /> match with snapshot 1`] = `
<div>
  <div
    class="sc-iBYQkv grSOXA"
  >
    <nav>
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-lmZTtQ-size-lg nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-OUZAS-zoomed-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-gEEnVX-ifAKUKQ-css"
        data-state="loading"
      >
        <span
          class="nextui-avatar-bg"
        />
        <img
          class="nextui-avatar-img nextui-avatar--loading"
          data-state="loading"
          src="/assets/ilustracion-rocio.png"
        />
      </span>
      <div>
        <ul>
          <li>
            <a
              href="/projects"
            >
              Projects
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/assets/cv-RocioBenitezGarcia.pdf"
            >
              CV
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
